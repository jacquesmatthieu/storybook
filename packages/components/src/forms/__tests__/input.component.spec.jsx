import React from 'react';
import { shallow } from 'enzyme';

import Input from '../input.component';

describe('Input', () => {
  let props;

  beforeEach(() => {
    props = {
      input: {
        name: 'foo',
      },
      meta: {},
    };
  });

  const getWrapper = () => shallow(<Input {...props} />);

  it('should render with default props', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with specified props', () => {
    // Given
    props.label = 'bar';
    props.meta = {
      error: true,
      touched: true,
      valid: true,
    };

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render multi line input', () => {
    // Given
    props.multiLine = true;
    props.rows = 3;

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
