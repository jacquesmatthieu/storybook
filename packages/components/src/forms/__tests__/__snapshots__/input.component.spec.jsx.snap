// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render multi line input 1`] = `
<InputGroup
  className=""
  fullWidth={false}
  label=""
  name="foo"
>
  <StyledTextArea
    error={false}
    fullWidth={false}
    id="foo"
    name="foo"
    rows={3}
    type="text"
  />
</InputGroup>
`;

exports[`Input should render with default props 1`] = `
<InputGroup
  className=""
  fullWidth={false}
  label=""
  name="foo"
>
  <inputcomponent__StyledInput
    error={false}
    fullWidth={false}
    id="foo"
    name="foo"
    type="text"
  />
</InputGroup>
`;

exports[`Input should render with specified props 1`] = `
<InputGroup
  className=""
  fullWidth={false}
  label="bar"
  name="foo"
>
  <inputcomponent__StyledInput
    error={true}
    fullWidth={false}
    id="foo"
    name="foo"
    type="text"
    valid={true}
  />
</InputGroup>
`;
